mender-update/update_module/v3/update_module.hpp

```diff
@@ -82,7 +82,9 @@ static std::string StateString[] = {
 	"Cleanup"};
 
 static inline std::string StateToString(State state) {
-	static_assert(sizeof(StateString) / sizeof(*StateString) == static_cast<int>(State::LastState));
+	static_assert(
+		sizeof(StateString) / sizeof(*StateString) == static_cast<int>(State::LastState),
+		"Make sure to keep State and StateString in sync!");
```

Comment:
Why not simply do `static std::string StateString[LastState + 1]` like I suggested before?

Reply:
Unfortunately it allows defining fewer strings than you have allocated, which means some entries could be unintentionally empty. So my check is stricter in that sense, because the count has to be exact.

Unrelated to that, I also realized that since this is `static` and in a header file, we are re-defining and re-storing it in the binary for each file that includes it. So I also removed `static` and moved the list and `StateToString` to the cpp file to save space.

Reply:
> Unfortunately it allows defining fewer strings than you have allocated, which means some entries could be unintentionally empty.

Huh? The rest is just silently initialized to `""` or what? No warnings at all?

Reply:
Nope, nothing. Just silent empty strings.